### ISSUE TITLE
Enum serializer is not picked-up

### ISSUE THREAD

**Issue by jerrinot** [hazelcast/hazelcast#15194]
The `EnumSerializer` is registered as a [constant](https://github.com/hazelcast/hazelcast/blob/de49652f82e8ab87154bc1a82803531dd1a0ab28/hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/SerializationServiceV1.java#L178) and this means there is no super-class lookup when searching for an interface. 

You can test this by throwing exception from `EnumSerializer` - the `EnumTest` is still passing! There is a silent fallback to Java serialization.

**Comment by ihsandemir** [comment]
@jerrinot Yes, it seems like the EnumSerializer actually never worked out of the box. 

I examined the implementation of the `EnumSerializer` and it is not efficient. In the client protocol for example, we decided that the enums will not be supported as builtin type but we will use `int` to transfer enums. Using a similar reasoning, I am suggesting that we remove this builtin support for enum (remove `EnumSerializer`).
